metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java

```diff
@@ -517,6 +355,81 @@ private static boolean pluginHasSuccessfulRecords(
+            .getWorkflowExecutionsOverview(Set.of(datasetId), Set.of(PluginStatus.FINISHED)
+                    , Set.of(PluginType.PUBLISH), null, null, pagination);
+
+    // Extract all (finished) publish plugins inversely sorted by started date (most recent first).
+    final List<Pair<IndexToPublishPlugin, WorkflowExecution>> publishOperations = new ArrayList<>();
+    executionsWithPublishOperations.getResults().stream().map(ExecutionDatasetPair::getExecution)
+            .forEach(execution -> execution.getMetisPlugins().stream()
+                    .filter(plugin -> plugin instanceof IndexToPublishPlugin)
+                    .map(plugin -> (IndexToPublishPlugin) plugin)
+                    .forEach(plugin -> publishOperations.add(new ImmutablePair<>(plugin, execution))));
+    final Comparator<Pair<? extends ExecutablePlugin<?>, WorkflowExecution>> comparator = Comparator
+            .comparing(pair -> pair.getLeft().getStartedDate());
+    publishOperations.sort(comparator.reversed());
+
+    // Compile a list of all associated harvests (that led to one of these publish operations).
+    // Note: we assume that workflows don't cross each other (i.e. an earlier publish cannot have a
+    // later harvest). We stop when we find a full harvest (the latest full harvest).
+    final Map<ExecutedMetisPluginId, PluginWithExecutionId<? extends ExecutablePlugin<?>>> resultHarvests = new HashMap<>();
+    for (Pair<IndexToPublishPlugin, WorkflowExecution> publishOperation : publishOperations) {
+
+      // If the publish is not available, we have detected an anomaly. We are done.
+      if (MetisPlugin.getDataStatus(publishOperation.getLeft()) != DataStatus.VALID &&
+              MetisPlugin.getDataStatus(publishOperation.getLeft()) != DataStatus.DEPRECATED) {
+        return Collections.emptyList();
+      }
+
+      // Get the root harvest and add it to the map.
+      final PluginWithExecutionId<? extends ExecutablePlugin<?>> rootHarvest = getRootAncestor(
+              new PluginWithExecutionId<>(publishOperation.getRight(), publishOperation.getLeft()));
+      resultHarvests.put(ExecutedMetisPluginId.forPlugin(rootHarvest.getPlugin()), rootHarvest);
+
+      // If the root harvest is not a harvest, we have detected an anomaly. We are done.
+      if (!HARVEST_PLUGIN_GROUP
+              .contains(rootHarvest.getPlugin().getPluginMetadata().getExecutablePluginType())) {
+        return Collections.emptyList();
+      }
+
+      // If the root harvest is a full harvest, we are done.
+      final boolean incrementalHarvest = (rootHarvest.getPlugin() instanceof OaipmhHarvestPlugin)
+              && ((OaipmhHarvestPlugin) rootHarvest.getPlugin()).getPluginMetadata()
+              .isIncrementalHarvest();
+      if (!incrementalHarvest) {
+        break;
+      }
+    }
+
+    // Done. Sort and return the harvests in the right order.
+    return resultHarvests.values().stream()
```

Comment:
Why are we ordering here?
Also they are already inversely ordered from the previous list

Reply:
You'll notice that they are in a map (which is just easier to use when filtering out duplicates). So they are not sorted at all.

Reply:
Sorry I think I didn't express myself properly, so maybe I'll try another way:
So there is a list which is first sorted publishOperations.
Then that list is iterated and there are keys inserted on the map basically in order since the publishOperations is sorted previously.
So if we simply change the type of map from HashMap to LinkedHashMap, that means that the map will already be sorted at the return statement right? (It's inversely sorted though so not sure if that matter, but in this case it doesn't matter if I'm not mistaken?)

Hope I expressed it better this time around:)

Reply:
OK, I replaced it with a LinkedHashMap.